src/app/helpers/context/context-decorators.ts

```diff
@@ -0,0 +1,39 @@
+import { ElementRef } from "@angular/core";
+import { ContextRequestEvent, UnknownContext } from "./context";
+
+export interface WithContext {
+  elementRef: ElementRef;
+  ngAfterViewInit?(): void;
+}
+
+/**
+ * You can use this decorator to bind a context subscription to a class property.
+ *
+ * @example
+ * ```ts
+ * class MyComponent {
+ *    @ContextSubscription("my-context")
+ *    public handleContextChange(newValue) {
+ *      console.debug({ newValue });
+ *    }
+ * }
+ * ```
+ */
+export function ContextSubscription(token: UnknownContext) {
+  return (target: WithContext, propertyKey: string) => {
+    const originalNgAfterViewInit = target.ngAfterViewInit;
+
+    target.ngAfterViewInit = function () {
+      // invoke the original ngAfterViewInit method if it exists
+      if (originalNgAfterViewInit) {
+        originalNgAfterViewInit();
+      }
+
+      const handler = target[propertyKey].bind(this);
```

Comment:
No, calling the property doesn't inherently trigger a change detection cycle

I have fixed it so that it requests a change detection cycle after a context has been provided